lib/parse-args.js

```diff
@@ -191,18 +188,13 @@ function buildYargs (withCommands = false) {
 function hideInstrumenterArgs (yargv) {
   let argv = process.argv.slice(1)
   argv = argv.slice(argv.indexOf(yargv._[0]))
-  if (argv[0][0] === '-') {
-    argv.unshift(process.execPath)
-  }
   return argv
 }
```

Comment:
What use case does this line address?  The only documentation I could find mentioning the `-` is omitting the double dash `--` for configurations that use [package.json or loaded configuration files.](https://github.com/bcoe/c8#cli-options--configuration)   Configuration files for C8 are included in another section of this file.  Does this block of code support loading [configurations via package.json](https://github.com/bcoe/c8#cli-options--configuration)?   Do I need to add a Test Case instead of removing the code? If not, what purpose does this serve?

As the function name suggests this function has something to do with instrumentation arguments.  The only reference I found regarding instrumentation in the documentation was the section [Source map files versus inline source maps.](https://github.com/bcoe/c8#source-map-files-versus-inline-source-maps)

From what I can tell, if the first character of the first argument is a `-` then add the path where C8 is executed to the end of the argument array.

Reply:
See the original implementation that this was pulled from:

https://github.com/istanbuljs/nyc/pull/350

I believe it has to do with the `--` operator, which indicates that parsing should stop? There's a test case you can pull over.

Reply:
Thank you for that pointer @bcoe.  I was able to derive a test case from those examples in that link and those lines were able to get covered.

That code section has to do with undocumented behavior on passing arguments to node.js via C8.  It maybe a backward compatibility issue with previous versions of nyc.